### PR TITLE
[SwiftASTContext] Fix GetTypeBitAlign to work in the general case

### PR DESCRIPTION
**Pull request by dcci**
In particular:
1) Generic types [which aren't currently mapped]
2) non-fixed layout types.

### CODE REVIEW

**Comment by dcci** [comment]
This is a WIP, more to come!

**Comment by dcci** [comment]
Down to 14 tests failure :)

File: source/Target/SwiftLanguageRuntime.cpp [hunk 1]
```diff
@@ -2594,6 +2594,12 @@ llvm::Optional<uint64_t> SwiftLanguageRuntime::GetByteStride(CompilerType type)
   return {};
 }
 
+llvm::Optional<size_t> SwiftLanguageRuntime::GetBitAlignment(CompilerType type) {
+  if (auto *type_info = GetTypeInfo(type))
+    return type_info->getAlignment();
```
Comment (adrian-prantl) [inline source/Target/SwiftLanguageRuntime.cpp:2599]: Why is this needed and why isn't it calling into SwiftASTContext::GetTypeBitAlign?

**Review by adrian-prantl — COMMENTED** [review]
Good work! One question inline.

File: source/Target/SwiftLanguageRuntime.cpp (hunk 1 above)
Comment (adrian-prantl) [reply source/Target/SwiftLanguageRuntime.cpp:2599]: Sorry, my mistake, this *is* the runtime call in the other function!

**Comment by dcci** [comment]
@swift-ci test and merge

**Review by bulbazord — APPROVED** [review]

**Comment by dcci** [comment]
@swift-ci test and merge
